src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -298,6 +298,15 @@ public final String getLine(int index) {
         return context.get().fileContents.getLine(index);
     }
 
+    /**
+     * Checks if the current file is a package-info.java file.
+     *
+     * @return true if the package file.
+     */
+    public final boolean inPackageInfo() {
```

Comment:
As to me too much details in abstract class, if we can not avoid checking file name, abstract class should have method to return name for Check to do anything required

Reply:
changed, see other pr https://github.com/checkstyle/checkstyle/pull/11502